server/src/main/java/org/apache/iotdb/db/metrics/server/ServerArgument.java

```diff
@@ -0,0 +1,347 @@
+			int wocidx = line.indexOf("WriteOperationCount");
+			long idletime = 0;
+			long kneltime = 0;
+			long usertime = 0;
+			while ((line = input.readLine()) != null) {
+				if (line.length() < wocidx) {
+					continue;
+				}
+				String caption = substring(line, capidx, cmdidx - 1).trim();
+				String cmd = substring(line, cmdidx, kmtidx - 1).trim();
+				if (cmd.indexOf("wmic.exe") >= 0) {
+					continue;
+				}
+				String s1 = substring(line, kmtidx, rocidx - 1).trim();
+				String s2 = substring(line, umtidx, wocidx - 1).trim();
+				List<String> digitS1 = new ArrayList<>();
+				List<String> digitS2 = new ArrayList<>();
+				digitS1.add(s1.replaceAll("\\D", ""));
+				digitS2.add(s2.replaceAll("\\D", ""));
+				if (caption.equals("System Idle Process") || caption.equals("System")) {
+					if (s1.length() > 0) {
+						if (!digitS1.get(0).equals("") && digitS1.get(0) != null) {
+							idletime += Long.valueOf(digitS1.get(0)).longValue();
+						}
+					}
+					if (s2.length() > 0) {
+						if (!digitS2.get(0).equals("") && digitS2.get(0) != null) {
+							idletime += Long.valueOf(digitS2.get(0)).longValue();
+						}
+					}
+					continue;
+				}
+				if (s1.length() > 0) {
+					if (!digitS1.get(0).equals("") && digitS1.get(0) != null) {
+						kneltime += Long.valueOf(digitS1.get(0)).longValue();
+					}
+				}
+				if (s2.length() > 0) {
+					if (!digitS2.get(0).equals("") && digitS2.get(0) != null) {
+						kneltime += Long.valueOf(digitS2.get(0)).longValue();
+					}
+				}
+			}
+			retn[0] = idletime;
+			retn[1] = kneltime + usertime;
+			return retn;
+		} catch (Exception ex) {
+			ex.printStackTrace();
+		} finally {
+			try {
+				proc.getInputStream().close();
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * read cpu info(linux)
+	 */
+	private long[] readLinuxCpu() {
+		long[] retn = new long[2];
+		BufferedReader buffer = null;
+		long idleCpuTime = 0;
+		long totalCpuTime = 0;
+		try {
+			buffer = new BufferedReader(new InputStreamReader(new FileInputStream("/proc/stat")));
+			String line = null;
+			while ((line = buffer.readLine()) != null) {
+				if (line.startsWith("cpu")) {
+					StringTokenizer tokenizer = new StringTokenizer(line);
+					List<String> temp = new ArrayList<String>();
+					while (tokenizer.hasMoreElements()) {
+						temp.add(tokenizer.nextToken());
+					}
+					idleCpuTime = Long.parseLong(temp.get(4));
+					totalCpuTime = Long.parseLong(temp.get(1)) + Long.parseLong(temp.get(2))
+							+ Long.parseLong(temp.get(3)) + Long.parseLong(temp.get(4));
+					break;
+				}
+			}
+			retn[0] = idleCpuTime;
+			retn[1] = totalCpuTime;
+		} catch (Exception e) {
+			e.printStackTrace();
+		} finally {
+			try {
+				buffer.close();
+			} catch (Exception e2) {
+				e2.printStackTrace();
+			}
+		}
+		return retn;
+	}
+
+	private String substring(String src, int start_idx, int end_idx) {
+		byte[] b = src.getBytes();

```

Comment:
This method is unnecessary, you can use Java API `src.substring(start_idx, end_idx)` to get the substring. What is different from your method is: `i >= start_idx && i < end_idx`.